### DISCUSSION TITLE
How to get the type for `JSXExpressionContainer` when creating a custom rule

### DISCUSSION THREAD

**Discussion by vinassefranche** [eslint/eslint#16652] [Q&A]
Hi everybody,

I'm trying to build a custom eslint rule that works using the rule listener `JSXExpressionContainer`. I'm using typescript and can't manage to have the type for the associated method. I can get the type of many other rules listeners like `AssignmentExpression` for example but everything that is react related is not present.
Here's the code I'm trying to have typescript compile:
```typescript
import type { Rule } from 'eslint';

export const myRule: Rule.RuleModule = {
  create: context => ({
    JSXExpressionContainer: node => {
      return context.report({ node, message: 'some message' });
    }
  }),
};
```
I'm having this error: `Parameter 'node' implicitly has an 'any' type.` which means that the type of the rule listener `JSXExpressionContainer` is not present in the `Rule` type.

Any idea how to fix this? Any module to import to add the types?

Thanks in advance!

**Comment by amareshsm** [comment]
For `node` try giving the type as `TSESTree.Node` (TSES - https://typescript-eslint.io/custom-rules#type-checking)

**Reply by vinassefranche** [reply]
Thanks for the answer @amareshsm 
I'll have a look at this lib that I did not know.
